modules/primo/src/Primo/BriefSearch/Facet.php

```diff
@@ -58,9 +60,19 @@ public function getValues() {
     $facetValueNodes = $this->xpath('//search:FACET[@NAME="' . $this->id . '"]/search:FACET_VALUES');
     foreach ($facetValueNodes as $facetValueNode) {
       /* @var \DOMNode $facetValueNode */
-      $keyAttribute = $facetValueNode->attributes->getNamedItem('KEY');
+      $keyAttributeValue = $facetValueNode->attributes->getNamedItem('KEY')->nodeValue;
+
+      // Map choose facets.
+      if ($this->id === 'lang' && !empty($mapped = ValueMapper::mapLanguageFromIso639($keyAttributeValue))) {
+        $keyAttributeValue = $mapped;
+      }
+      elseif ($this->id === 'genre' && !empty($mapped = ValueMapper::mapGenreFromCode($keyAttributeValue))) {
+        $keyAttributeValue = $mapped;
+      }
+
+
       $valueAttribute = $facetValueNode->attributes->getNamedItem('VALUE');
-      $values[$keyAttribute->nodeValue] = $valueAttribute->nodeValue;
+      $values[$keyAttributeValue] = $valueAttribute->nodeValue;
```

Comment:
This intertwines `Primo\BriefSearch` with the `ValueMapper` and through this `t()`. Is that necessary or could we move the mapping the to the `Primo\Ting` code?

Perhaps we can deal with this as a part of a refactoring.

Reply:
Ah, yes, you have a point there...

I've done a quick initial refactor and moved it to ```Result->getFacets()``` in ```Primo\Ting```